### ISSUE TITLE
incorrect constructors if type is a varname

### ISSUE THREAD

**Issue by Mr-Yann** [jameskermode/f90wrap#30]
With a derived type containing a variable called type, f90wrap produces __get__type_bn and __set__type_bn.
f2py3 then has the error : 'type_bn' at (1) is not a member of the structure.
Minimal example f90 file :

```fortran
module module_structure

type type_face
    integer :: type
end type type_face

end module module_structure
```

**Comment by jameskermode** [comment]
``type`` is a reserved word in Python, hence the remapping to ``type_bn`` (bn stands for "bad name"), which is the same transformation that the standard `f2py` would apply. It appears that this is not being done completely consistently, however. I'll look into it.

**Comment by Mr-Yann** [comment]
I fully agree that using type as a name while it is already used for something else even in fortran is a poor choice. I guess that most of the problems f90wrap will encounter with the code i try to use will come from non-rigorous subroutines. To see this on the bright side, it will give you leads to achieve a more robust wrapper ;)
____
Yann

**Comment by jameskermode** [comment]
Should now be fixed. Let me know if you have further problems.

**Comment by Mr-Yann** [comment]
It seems to work and I thank you for your responsiveness.
I am now faced with another issue : the fortran code I try to use is 
full of "pointer" variables like this :

real,dimension(:,:),pointer :: x

 From what i read about f2py, pointer data is not supported. Could your 
wrapper do something about that, or at least is there any way around 
this "pointer" property of the variable ?
For now, subroutines with pointer variables are ignored without warning 
and the module class ends up with only "pass" in the user-friendly .py 
generated file.

Thanks again for your time an dedication to your project : )

_____
Yann

**Comment by jameskermode** [comment]
I don't remember where the limitation on pointer arugments comes wrong. I will try to have a look at a simple example soon-  I'm reopening this issue to keep track.

**Comment by MrYann** [comment]
Here is a minimal example producing an empty python wrapper and an f2py3 error at the Building Modules step:

module module_calcul
contains
subroutine evaluer_valeur(x)
real,dimension(:,:),pointer :: x
	x = 2
	return
end subroutine evaluer_valeur
end module module_calcul

**Comment by MrYann** [comment]
I'm sorry to re-open this case but it seems that the fix for "type" as a member name inside a type is not 100% working. With the minimal example provided at the beginning of this issue, the python commands

    import maxwell
    a = maxwell.module_structure.Face()
    
results in : AttributeError: module '_maxwell' has no attribute 'f90wrap_face_initialise'.

**Comment by MrYann** [comment]
Got it.
I propose you to replace line 714 of file f90wrapgen.py
       if localvar in getattr(el, "uses", []) or localvar in extra_uses:
by
       if localvar in getattr(el, "uses", []) or localvar in extra_uses or localvar == "type":

I know you have a list of bad names, so of course it would be better to modify with something like
     ... localvar not in bad_names_list
and this should do the trick.

**Comment by MrYann** [comment]
Hi : )

Rather than checking for uses, bad names, or who knows what we are forgetting this time, I think it is easier to just append "_f90wrap" to all type members. I updated my pull request accordingly.

(and proposed a bug fix on another subject as well)
